WebApplication/Backend/Repositorys/SurveyRepository.cs

```diff
@@ -0,0 +1,428 @@
+                statistics[6].AverageRating += Double.Parse(s.Question11);
+                statistics[6].increment(Int32.Parse(s.Question11));
+                statistics[7].AverageRating += Double.Parse(s.Question12);
+                statistics[7].increment(Int32.Parse(s.Question12));
+                statistics[8].AverageRating += Double.Parse(s.Question13);
+                statistics[8].increment(Int32.Parse(s.Question13));
+                statistics[9].AverageRating += Double.Parse(s.Question14);
+                statistics[9].increment(Int32.Parse(s.Question14));
+                statistics[10].AverageRating += Double.Parse(s.Question15);
+                statistics[10].increment(Int32.Parse(s.Question15));
+                statistics[11].AverageRating += Double.Parse(s.Question16);
+                statistics[11].increment(Int32.Parse(s.Question16));
+                statistics[12].AverageRating += Double.Parse(s.Question17);
+                statistics[12].increment(Int32.Parse(s.Question17));
+                statistics[13].AverageRating += Double.Parse(s.Question18);
+                statistics[13].increment(Int32.Parse(s.Question18));
+                statistics[14].AverageRating += Double.Parse(s.Question19);
+                statistics[14].increment(Int32.Parse(s.Question19));
+                statistics[15].AverageRating += Double.Parse(s.Question20);
+                statistics[15].increment(Int32.Parse(s.Question20));
+                statistics[16].AverageRating += Double.Parse(s.Question21);
+                statistics[16].increment(Int32.Parse(s.Question21));
+                statistics[17].AverageRating += Double.Parse(s.Question22);
+                statistics[17].increment(Int32.Parse(s.Question22));
+                statistics[18].AverageRating += Double.Parse(s.Question23);
+                statistics[18].increment(Int32.Parse(s.Question23));
+            }
+            for (int i = 0; i < 19; i++)
+            {
+                statistics[i].generatePercents();
+                statistics[i].AverageRating = statistics[i].AverageRating / reports.Count;
+            }
+            return round2Decimals(statistics);
+        }
+
+        ////Aleksa Repovic RA52/2017
+        /// <summary>
+        ///calculates statistics for each topic
+        ///</summary>
+        ///<returns>
+        ///list of objects containing information about each topic
+        ///</returns>
+        public List<StatisticAuxilaryClass> getStatisticsEachTopic()
+        {
+
+            List<Survey> reports = new List<Survey>();
+            reports = GetSurveys("Select * from surveys");
+
+            List<StatisticAuxilaryClass> statistics = new List<StatisticAuxilaryClass>();
+            for (int i = 0; i < 5; i++)
+            {
+                statistics.Add(new StatisticAuxilaryClass());
+            }
+
+            foreach (Survey s in reports)
+            {
+                statistics[0].AverageRating += Double.Parse(s.Question5);
```

Comment:
I have no idea what is going on here but I think you can extract something to a method